contracts/HydraChain/modules/Inspector/Inspector.sol

```diff
@@ -38,14 +43,46 @@ abstract contract Inspector is IInspector, ValidatorManager {
 
     // _______________ External functions _______________
 
+    /**
+     * @inheritdoc IInspector
+     */
+    function initiateBan(address validator) external {
+        if (bansInitiated[validator] != 0) {
+            revert BanAlreadyInitiated();
+        }
+
+        if (!isSubjectToInitiateBan(validator)) {
+            revert NoInitiateBanSubject();
+        }
+
+        bansInitiated[validator] = block.timestamp;
+        hydraStakingContract.temporaryRemove(validator);
+    }
+
+    /**
+     * @inheritdoc IInspector
+     */
+    function terminateBanProcedure() external {
+        if (bansInitiated[msg.sender] == 0) {
+            revert NoBanInititated();
+        }
+
+        bansInitiated[msg.sender] = 0;
+        hydraStakingContract.returnBack(msg.sender);
```

Comment:
Could we change the function name so it is clear what we returnBack, same for temporaryRemove function.
Make - disableStaker/Validator - enable/recoverStaker/Val